File: 002.rb
#!/usr/bin/env ruby -wKU

fibs = [1, 1]
ans = 0
while fibs[0] < 4e6
  ans += fibs[0] if fibs[0].even?
  *fibs = fibs[1], fibs[0] + fibs[1]
end
puts ans

File: Python/002.py
#!/usr/bin/env python
# encoding: utf-8


def main():
    """
    calculate only the even value in fibonacci seq
    """
    ans = 0
    a, b = 2, 0
    while a < 4000000:
        ans += a
        a, b = 4*a+b, a
    print(ans)

if __name__ == '__main__':
    main()
